[STT-1301] Update filters and UI config to work with new metadata fields

eos87 opened the pull request:
- Updated `ui_config.json` and `settings.py` to use the `service` field instead of `sttdepartment` UI display and filtering. Adjusted metadata fields, group definitions, and aggregations to reflect this change, and updated related components and schemes accordingly. 
- Replaced `sttsubj` with `topics` in ui config too
- Adjusted `service` field to use the new component implemented in https://github.com/superdesk/newsroom-core/pull/1357

Depends on https://github.com/superdesk/newsroom-core/pull/1356, https://github.com/superdesk/newsroom-core/pull/1357 and https://github.com/superdesk/newsroom-app-stt/pull/244

Resolves: [STT-1301]

[STT-1301]: https://sofab.atlassian.net/browse/STT-1301?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

eos87 committed:
Update filters and UI config to work with new metadata fields

Updated `ui_config.json` and `settings.py` to use the `service` field instead of `sttdepartment` for department-related metadata and grouping. Adjusted metadata fields, group definitions, and aggregations to reflect this change, and updated related components and schemes accordingly.
Replaced `sttsubj` with `mediatopic` in ui config too

STT-1301

petrjasek commented on server/data/ui_config.json:
```diff
@@ -1,64 +1,96 @@
+            },
+            "subject" : {
+                "scheme" : [
+                    "mediatopic"
```
should be `topics`

petrjasek commented on server/data/ui_config.json:
```diff
@@ -1,64 +1,96 @@
+                "urgency",
+                "source",
+                {
+                    "field": "service",
```
I wonder if we should have it implemented in a way that you would just put there `service` and the component would be picked automatically, maybe with some option for bold. in general I don't think there will be much use to show certain field with different components, it will be mostly `1:1` mapping

eos87 replied:
It makes sense. I made the changes to simplify it. I also updated https://github.com/superdesk/newsroom-core/pull/1357 so service is mapped directly.

eos87 committed:
Simplify service field config in ui_config.json

Replaces complex service field objects with simple string references to 'service'. This removes custom component and style definitions.

STT-1301

eos87 committed:
Update subject scheme from `mediatopic` to `topics`

STT-1301

eos87 commented:
Hi @petrjasek, please take another look when you have some time. This PR now also depends on https://github.com/superdesk/newsroom-app-stt/pull/244

petrjasek committed:
Merge remote-tracking branch 'origin/next' into hg/STT-1301-update-filters-templates

petrjasek commented on server/settings.py:
```diff
@@ -38,23 +38,16 @@
             "parent": "subject",
             "field": "scheme",
-            "value": "sttsubj",
+            "value": "mediatopic",
```
this one should be `topics`

eos87 committed:
Update settings.py

eos87 committed:
Merge remote-tracking branch 'origin/next' into hg/STT-1301-update-filters-templates
